app/src/org/commcare/android/analytics/TimedStatsTracker.java

```diff
@@ -0,0 +1,112 @@
+package org.commcare.android.analytics;
+
+import android.content.SharedPreferences;
+import android.util.Log;
+
+import org.commcare.android.javarosa.AndroidLogger;
+import org.commcare.dalvik.application.CommCareApp;
+import org.commcare.dalvik.application.CommCareApplication;
+import org.javarosa.core.services.Logger;
+import org.joda.time.DateTime;
+
+import java.util.Date;
+
+/**
+ * Utils for keeping track of certain app events and then reporting them to google analytics
+ *
+ * @author amstone
+ */
+public class TimedStatsTracker {
+
+    private static final String TAG = TimedStatsTracker.class.getSimpleName();
+
+    private static String KEY_LAST_FORM_NAME = "last-form-name";
+    private static String KEY_LAST_FORM_START_TIME = "last-form-start-time";
+    private static String KEY_LAST_SESSION_START_TIME = "last-session-start-time";
+    private static String KEY_LAST_LOGGED_IN_USER = "last-logged-in-user";
+
+    public static void registerEnterForm(String formTitle) {
+        SharedPreferences.Editor editor =
+                CommCareApplication._().getCurrentApp().getAppPreferences().edit();
+        editor.putString(KEY_LAST_FORM_NAME, formTitle).
+                putLong(KEY_LAST_FORM_START_TIME, currentTime());
+        editor.commit();
```

Comment:
Might be nice to chain the `editor.commit()` call to the previous line for consistency.
Also, I'm a fan of putting the `.` on the new line with the chained method call, because that offers an additional piece of visual info to distinguish the chained call from a normal method invocation.